src/components/CategoryList.jsx

```diff
@@ -0,0 +1,199 @@
+import React, { useMemo, useState, memo } from "react";
+import { FiChevronDown, FiChevronRight, FiEdit } from "react-icons/fi";
+import { MdDelete } from "react-icons/md";
+
+function CategoryList({
+  categoryName, // "Main" | "Side" | "Dessert" | "Beverage" | "Miscellaneous"
+  items = [], // array of event items for this category
+  categoryColor = "text-gray-800",
+  dietaryIcons = {}, // { vegan:{icon,color}, ... }
+  onEditItem, // (item) => void
+  onDeleteItem, // (item) => void
+  defaultExpanded = true,
+}) {
+  const [isExpanded, setIsExpanded] = useState(defaultExpanded);
+  const [expandedItems, setExpandedItems] = useState(() => new Set());
+
+  const toggleItem = (id) =>
+    setExpandedItems((prev) => {
+      const s = new Set(prev);
+      s.has(id) ? s.delete(id) : s.add(id);
+      return s;
+    });
+
+  // Mirror original behavior: sort items by title
+  const sorted = useMemo(
+    () => [...items].sort((a, b) => a.title.localeCompare(b.title)),
+    [items],
+  );
+
+  const headerText =
+    categoryName === "Miscellaneous" ? categoryName : `${categoryName}s`;
+
+  return (
+    <div className="mb-6">
+      {/* Header */}
+      <div
+        className="mb-3 flex cursor-pointer items-center justify-between"
+        onClick={() => setIsExpanded((v) => !v)}
+      >
+        <h3 className={`text-lg font-bold ${categoryColor}`}>
+          {headerText} ({items.length})
+        </h3>
+        {isExpanded ? (
+          <FiChevronDown size={24} className={categoryColor} />
+        ) : (
+          <FiChevronRight size={24} className={categoryColor} />
+        )}
+      </div>
+
+      {/* Items */}
+      {isExpanded && sorted.length > 0 && (
```

Comment:
🔧 I would like variable naming to be more explicit here. I think sorted is too vague. 
```suggestion
      {isExpanded && sortedItems.length > 0 && (
```